src/librustc_mir/const_eval.rs

```diff
@@ -647,10 +647,15 @@ pub fn const_eval_raw_provider<'a, 'tcx>(
                                                    "could not evaluate static initializer");
             // Ensure that if the above error was either `TooGeneric` or `Reported`
             // an error must be reported.
-            if tcx.sess.err_count() == 0 {
-                tcx.sess.delay_span_bug(err.span,
+            let tracked_err = tcx.sess.track_errors(|| {
+                err.report_as_error(ecx.tcx,
```

Comment:
You should remove the `report_as_error` call above. I'm not sure what types `tracked_err` has, but the `Ok` path should give you the same value that you got previously in `reported_err`. What's the value in the `Err` part?

Reply:
Please let me ask you two questions.

1. What should I place instead of `report_as_error`?
2. How I check if `track_err` returns `Ok`?

Reply:
for 2. you are already checking that via the `match` below. You can pattern match on the `Ok` value via `Ok(reported_err)`.

for 1. nothing, you already are calling it inside of the `track_errors` call, so you are replacing it with the `tcx.sess.track_errors(|| { err.report_as_error ....`

Reply:
Okay, thank you! And what should I do in the `Err` part?

Reply:
I'm not sure, that depends on what is returned there. Have a look at the return type and emit an apropriate `ErrorHandled` enum variant

Reply:
I pushed WIP commit. Test will be failed because types are incompatible in `match`. 
But I want to know if my work is correct so far. Could you check this?

Reply:
Yes, this is correct so far.

In the `Ok` arm, the return value should be `v` I believe. The `Err` pattern can probably be `Err(ErrorReported)` and the arm value `ErrorHandled::Reported` (or whatever the correct variant is).

Reply:
I modified as follows:

```Rust
match reported_err {
    Ok(v) => {
        tcx.sess.delay_span_bug(err.span,
                            &format!("static eval failure did not emit an error: {:#?}",
                            v));
        v
    },
    Err(ErrorReported) => ErrorReported,
}
```

but `()` is returned in the `Ok` arm and `rustc::util::common::ErrorReported` is returned in the `Err` arm, so the compile is failed. Any solutions?